inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpChannelHandler.java

```diff
@@ -223,41 +208,59 @@ private void responsePackage(ResultCode code, MessageEvent e)
      * exceptionCaught
      * 
      * @param  ctx
-     * @param  e
+     * @param  cause
      * @throws Exception
      */
     @Override
-    public void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) throws Exception {
-        LOG.error("exception caught", e.getCause());
-        super.exceptionCaught(ctx, e);
-        if (e.getChannel() != null) {
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
+        LOG.error("exception caught cause = {}", cause);
+        if (ctx.channel() != null) {
             try {
-                e.getChannel().disconnect();
-                e.getChannel().close();
+                ctx.fireExceptionCaught(cause);
```

Comment:
If ctx.fireExceptionCaught get an Exception, the operation of disconnect and close will not be invoked.

Reply:
done